platform/core/src/feature_file_system.rs

```diff
@@ -0,0 +1,495 @@
+    Ok(fs_prefix + "/" + id)
+}
+
+#[allow(dead_code)]
+fn fs_path_from_resource_url(
+    resource_url: &ResourceUrl,
+    config: &impl FeatureFSConfigTrait,
+) -> Result<String, CoreFailure> {
+    let fs_prefix = feature_files_path(config)?;
+    let res_prefix = "polypod://FeatureFiles".to_string();
+    swap_prefix(resource_url, &res_prefix, &fs_prefix).map_err(|err| {
+        CoreFailure::failed_to_convert_to_fs_path_from_resource_url(resource_url.to_string(), err)
+    })
+}
+
+#[allow(dead_code)]
+fn resource_url_from_fs_path(
+    fs_path: &str,
+    config: &impl FeatureFSConfigTrait,
+) -> Result<String, CoreFailure> {
+    let fs_prefix = feature_files_path(config)?;
+    let res_prefix = "polypod://FeatureFiles/".to_string();
+    swap_prefix(fs_path, &fs_prefix, &res_prefix).map_err(|err| {
+        CoreFailure::failed_to_convert_to_resource_url_from_fs_path(fs_path.to_string(), err)
+    })
+}
+
+fn swap_prefix(string: &str, from: &str, to: &str) -> Result<String, String> {
+    if !string.starts_with(from) {
+        return Err(format!("{} does not start with {}", string, from));
+    }
+    Ok(string.replace(from, to))
+}
+
+#[allow(dead_code)]
+fn make_sure_feature_files_dir_exists(
+    platform_fs: &impl FileSystem,
+    config: &impl FeatureFSConfigTrait,
+) -> Result<(), CoreFailure> {
+    let files_path = feature_files_path(config)?;
+    if platform_fs.exists(&files_path) {
+        platform_fs.create_dir_structure(&files_path)?;
+    }
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn import(
+    url: &Url,
+    dest_resource_url: Option<ResourceUrl>,
+    platform_fs: &impl FileSystem,
+    config: &impl FeatureFSConfigTrait,
+) -> Result<ResourceUrl, CoreFailure> {
+    make_sure_feature_files_dir_exists(platform_fs, config)?;
+
+    let fs_path = match dest_resource_url {
+        Some(res_url) => fs_path_from_resource_url(&res_url, config),
+        None => fs_path_from_id(&Uuid::new_v4().to_string(), config),
+    }?;
+
+    platform_fs.unzip(url.as_str(), &fs_path)?;
+
+    resource_url_from_fs_path(&fs_path, config)
+}
+
+#[allow(dead_code)]
+struct Metadata {
+    is_directory: bool,
+    size: String,
+    time: String,
+    name: String,
+    id: String,
+}
+
+#[allow(dead_code)]
+fn metadata(
+    resource_url: &ResourceUrl,
+    platform_fs: &impl FileSystem,
+    config: &impl FeatureFSConfigTrait,
+) -> Result<Metadata, CoreFailure> {
+    let fs_path = fs_path_from_resource_url(resource_url, config)?;
+    let is_dir = platform_fs.is_directory(&fs_path);
+    let size = platform_fs.size(&fs_path)?;
+    let time_mod = platform_fs.time_modified(&fs_path)?;
+    let name = platform_fs.name(&fs_path)?;
+    Ok(Metadata {
+        is_directory: is_dir,
+        size,
+        time: time_mod,
+        name,
+        id: fs_path,
+    })
+}
+
+#[allow(dead_code)]
+fn read_dir(
+    resource_url: &ResourceUrl,
+    platform_fs: &impl FileSystem,
+    config: &impl FeatureFSConfigTrait,
```

Comment:
Here are a couple of functions with the same three arguments. Wouldn't it make sense to turn those arguments into a struct, and bind these as methods to them?